extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientProducer.java

```diff
@@ -9,9 +9,11 @@
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.quarkus.arc.DefaultBean;
 
-@ApplicationScoped
+@Singleton
 public class KubernetesClientProducer {
 
+    private KubernetesClient client;
```

Comment:
No biggie, but instead of storing it in an attribute to destroy later, you can pass it as a method argument.

Eg. 
```java
@PreDestroy
public void destroy(KubernetesClient client) { 
   client.close();
}
```

Reply:
Ah ignore me. I am confusing it with CDI's `@Destroyed` (https://docs.jboss.org/cdi/api/2.0/javax/enterprise/context/Destroyed.html)

Reply:
looks better, and avoids testing for null.